ui/console/src/main/webapp/scripts/controllers/realm-selector-controller.ts

```diff
@@ -0,0 +1,44 @@
+/// <reference path="../../vendor/vendor.d.ts" />
+
+module Controllers {
+
+    'use strict';
+    export class SelectRealmController {
+        public static $inject = ['$modal', '$log', 'Auth'];
+        constructor(private $modal:any, private $log:ng.ILogService, private Auth:Services.AuthService) {
+            if (Auth.isAuthenticated()) return;
+
+            var modalInstance = $modal.open({
+                templateUrl: 'views/select-realm-modal.html',
+                controller: 'SelectRealmModalInstanceController as modalInstance',
+                keyboard: false,
+                backdrop: 'static'
+            });
+
+            modalInstance.result.then(function (selectedRealm) {
+                $log.info('Realm to authenticate against: ' + selectedRealm);
+            });
+        }
+
+    }
+
+    export class SelectRealmModalInstanceController {
+        public static $inject = ['$modalInstance', 'Auth'];
+        constructor(private $modalInstance:any, private Auth:Services.AuthService) {
+        }
+
+        realms = ['acme-other-affairs', 'acme-roadrunner-affairs'];
```

Comment:
Just realized that this wasn't removed/changed before sending the PR! This needs to be externalized.